source/m2mbase.cpp

```diff
@@ -84,6 +88,15 @@ M2MBase::M2MBase(const M2MBase& other) :
     _is_under_observation(other._is_under_observation),
     _function_pointer(NULL)
 {
+    _resource_type = alloc_copy(other._resource_type, other._resource_type_length);
+    _resource_type_length = other._resource_type_length;
+    _uri_path = alloc_copy(other._uri_path, other._uri_path_length);
+    _uri_path_length = other._uri_path_length;
+    _interface_description = alloc_copy(other._interface_description, other._uri_path_length);
+    _interface_description_length = other._interface_description_length;
+    _name = other._name;
```

Comment:
Is this new _name assignment really needed here, or is it part of the name-related code which is mostly in comments?